third_party/tlx/language/mem_ops.py

```diff
@@ -58,36 +58,89 @@ def local_alloc(
     full_shape = [unwrapped_num] + unwrapped_shape
     dtype = tl._unwrap_if_constexpr(dtype)
     elem_type = dtype.to_ir(_semantic.builder)
-    if layout is None:
-        if storage == tlx.storage_kind.smem:
-            layout = tlx.nv_mma_shared_layout_encoding(
-                    shape=shape, 
-                    order=[1, 0], 
-                    elemType=dtype,
-                    numCTAsPerCGA=[1, 1], 
-                    numCTASplit=[1, 1], 
-                    numCTAOrder=[1, 1],
-                    fp4Padded=False)
+
+    if storage == tlx.storage_kind.smem:
+        # if layout is given by user, continue to create layout_handle
+        # otherwise, create default NVMMA for 2D tensor and swizzled_shared for others, then continue to create layout_handle
+        if isinstance(layout, tlx.nv_mma_shared_layout_encoding) or (layout is None and len(shape) == 2):
+            layout = layout if layout else tlx.nv_mma_shared_layout_encoding.make_default(shape, dtype)
             layout_handle = _semantic.builder.make_nv_mma_shared_encoding_attr(
-                [int(x) for x in layout.shape],
-                layout.order,
-                layout.elemType.to_ir(_semantic.builder),
-                layout.numCTAsPerCGA,
-                layout.numCTASplit,
-                layout.numCTAOrder,
-                layout.fp4Padded,
-            )
-        else:
-            layout = tlx.tensor_memory_layout_encoding.make_default(shape)
-            layout_handle = _semantic.builder.make_tensor_memory_encoding_attr(
+                    [int(x) for x in layout.shape],
+                    layout.order,
+                    layout.elemType.to_ir(_semantic.builder),
+                    layout.numCTAsPerCGA,
+                    layout.numCTASplit,
+                    layout.numCTAOrder,
+                    layout.fp4Padded,
+                )
+        elif isinstance(layout, tlx.swizzled_shared_layout_encoding) or (layout is None and len(shape) == 1):
```

Comment:
Let's fall back to default swizzled layout for 1D tensor alloc. 1D TMA load should still require 1D MMA layout and the layout propagation should get that. 1D non-TMA async load should require an optimal than default swizzled layout, and the layout propagation is supposed to get that too.

Reply:
Similar fix is also included in https://github.com/facebookexperimental/triton/pull/223